### PR TITLE
[Merged by Bors] - Add transparency examples

### LINKED ISSUE #3215: Add example demonstrating transparency
## How can Bevy's documentation be improved?

Setting objects to be transparent is a common task, but doing so is not immediately obvious (see #64). We should demonstrate this in:

1. A 2D context.
2. A 3D context.
3. A UI context.

Ideally with relatively simple but still verifiable examples.

### PR DESCRIPTION
**Pull request by Wcubed**
Adds examples demonstrating transparency for 2d, 3d and UI.

Fixes #3215.

### CODE REVIEW

File: examples/README.md
Comment (mockersf) [inline examples/README.md:94]: could you remove those whitespace changes (and the extra `-`)?

Comment (Wcubed) [reply examples/README.md]: Done 👍🏻. Apparently my IDE tried to prettify the table 😅.

**Review by alice-i-cecile — APPROVED** [review]

File: examples/ui/transparency_ui.rs
Comment (alice-i-cecile) [inline examples/ui/transparency_ui.rs:45]:
```suggestion
    // to demonstrate the text looks different due to its transparency.
```

**Comment by alice-i-cecile** [comment]
Excellent work; these are simple, clear and minimal.

**Comment by alice-i-cecile** [comment]
Removing from the milestone. This PR is great, but we need to assess how we display the growing number of examples we want to have 😅

File: examples/3d/transparency_3d.rs
Comment (superdump) [inline examples/3d/transparency_3d.rs:32]: It would be good to demonstrate `AlphaMode::Mask` as well.

**Review by superdump — CHANGES_REQUESTED** [review]
Missing an example case that would be good to demonstrate and needs updating but LGTM after that.

**Comment by Nilirad** [comment]
This PR adds new examples. Adding module and item level doc comments, as described in:

- #3951 and
- #4438,

would be really useful to those who will browse examples.

**Comment by superdump** [comment]
@Wcubed any chance you could update this PR and address the few remaining pieces of review feedback? Then we can get it merged. :)

**Comment by IceSentry** [comment]
@Wcubed now that #4658 is merged, you could show setting the alpha channel and relying on the From<Color> impl to set the alpha_mode. I would still keep the full example, but showing the shorthand would be nice.

**Review by dyc3 — APPROVED** [review]
These will be great to have!

**Review by alice-i-cecile — APPROVED** [review]

**Review by IceSentry — APPROVED** [review]

**Review by superdump — APPROVED** [review]
We can merge this when the conflicts are resolved, imo.

**Comment by Wcubed** [comment]
Rebased and resolved the conflicts in `Readme.md` 👍🏼

**Pull request closed without merging**
